src/D2L.Bmx/Okta/OktaClient.cs

```diff
@@ -1,62 +1,72 @@
 using System.Net;
-using System.Net.Http.Headers;
 using System.Net.Http.Json;
 using System.Text.Json;
 using D2L.Bmx.Okta.Models;
 
 namespace D2L.Bmx.Okta;
 
-internal interface IOktaApi {
-	void SetOrganization( string organization );
-	void AddSession( string sessionId );
+internal interface IOktaClientFactory {
+	IOktaAnonymousClient CreateAnonymousClient( string org );
+	IOktaAuthenticatedClient CreateAuthenticatedClient( string org, string sessionId );
+}
```

Comment:
The most problematic methods in the original `OktaApi` class were `SetOrganization` & `AddSession`, as they are the ones manipulating internal state and has impact on when you can call which other methods.
Pulling them out into a factory (and hence naturally separating the anonymous & authed states) is the key to making the new classes stateless.

The new factory interface/class is also the key to understand the structure of the new code.
The rest is pretty mundane copy/paste.